mypyc/test/test_run.py

```diff
@@ -182,6 +183,9 @@ def run_case(self, testcase: DataDrivenTestCase) -> None:
                     print(line)
                 assert False, 'Compile error'
 
+            # Check that serialization works on this IR
+            check_serialization_roundtrip(ir)
```

Comment:
Nice, this is a clever way to test serialization -- without writing any new tests!

Have you checked that the existing tests cover all the interesting cases? What about things like multiple SCCs that depend on each other in different ways.

Reply:
Well, it catches all of the interesting cases that I know we had bugs in ;).
(My original take only ran this test on genops tests, since run wasn't set up in a way that was conducive to it, but I hacked it in to see if it would help and it found bugs that testing with genops didn't, so I did #7731 to enable the test on run.)

Our run tests are pretty decent I think at covering most of what mypyc does.

It definitely doesn't really test any SCC stuff, because the code that drives deserialization an SCC at a time hasn't been written yet (well, it has, but not as part of this PR) so it is definitely not tested :P. This test just deserializes everything at once.